audb/core/publish.py

```diff
@@ -679,6 +684,28 @@ def publish(
         verbose=verbose,
     )
 
+    # ensure table and attachment IDs
+    # contain only chars allowed by the backend
+    # as the IDs are included in the filenames of the archives
+    # uploaded to the backend
+    allowed_chars = audbackend.core.utils.BACKEND_ALLOWED_CHARS
```

Comment:
Why don't we simply rely on the error raised by `audbackend`?

Reply:
Because otherwise parts of the database will be uploaded already?

Reply:
If so, would it make sense to make the utility function in `audbackend` available rather than repeating all the code here?

Reply:
There is also another difference as we have to forbid `/` as char here.

Reply:
Otherwise, most of the code is just about providing explicit error messages for table and attachment IDs.

Reply:
Ok, I see. Btw: do we already check the version string somewhere?

Reply:
Otherwise uploading may also fail once https://github.com/audeering/audbackend/issues/101#issuecomment-1521653330 has been implemented. But we probably want to be more strict here and enforce `audeer.is_semantic_version()`.

Reply:
We use [audeer.LooseVersion](https://audeering.github.io/audeer/api/audeer.LooseVersion.html) to convert the given version string to a version. The code seems not able to fail, but it might ignore some parts of the version string:
https://github.com/audeering/audeer/blob/2fa3e10790002db079773fa97a71b284a19cf06d/audeer/core/version.py#L324-L348

If you think we should change this (e.g. by switching to `audeer.StrictVersion` which raises `ValueError` for not supported versions), please open another issue as this is not related to the topic of this pull request.

Reply:
If a version string can pass that contains any of the characters we forbid in the PR, I guess we need some action yes.

Reply:
I created https://github.com/audeering/audb/issues/292